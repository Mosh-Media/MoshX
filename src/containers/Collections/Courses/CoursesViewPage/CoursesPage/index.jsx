/**
 * Collections is a container that fetches firebase data using hooks and renders a list of all collections
 * @param {Object} authUser - Passed from parent container and has everything about the logged in user
 * @param {Object} collection - Passed from parent with a title and a path of the collection
 * @param {Object} firebase - Firebase class provides access to authUser and db - comes from withAuthentication hoc
 * @param {Object} match - Contains information about how a <Route path> matched the URL - comes from withRouter and passed to withAuthentication hoc
 * @returns {<CoursePage/>} - returns CoursePage component which contains the rest of the components
 */

import React, { lazy } from "react";

import Footer from "components/shared/Footer";
import MoPage from "components/library/MoPage";

import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import Grow from "@material-ui/core/Grow";
import Navigation from "components/shared/Navigation";
import SignUpCard from "components/shared/CardList/SignUpCard";
import NewItemCard from "components/shared/CardList/NewItemCard";

const CardList = lazy(() => import("components/shared/CardList"));

const CoursesPage = ({
  authUser,
  collectionDetails,
  courses,
  firebase,
  isLoading,
  isAdmin,
  itemOptions
}) => {
  return (
    <Container maxWidth="lg">
      <Navigation authUser={authUser} firebase={firebase} />
      <MoPage title={collectionDetails?.title}>
        <Grow
          in={!isLoading}
          mountOnEnter
          timeout={{ enter: 400, exit: 400 }}
          unmountOnExit
        >
          <Grid container spacing={4} alignItems="center">
            <CardList
              isAdmin={isAdmin}
              items={courses}
              itemOptions={itemOptions}
            />
            {!!isAdmin && !courses?.isLoading && (
              <Grid item xs={12} sm={12} md={4} lg={4}>
                <NewItemCard
                  type="new"
                  title={itemOptions?.newItem?.title}
                  url={itemOptions?.newItem?.url}
                />
              </Grid>
            )}
            {!itemOptions?.authUser && !courses?.isLoading && (
              <Grid item xs={12} sm={12} md={4} lg={4}>
                <SignUpCard />
              </Grid>
            )}
          </Grid>
        </Grow>
        <Footer />
      </MoPage>
    </Container>
  );
};

export default CoursesPage;
