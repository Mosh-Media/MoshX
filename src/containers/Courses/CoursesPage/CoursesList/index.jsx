/**
 * Loops over lsit of courses and renders courses with category title
 * @param {Object} authUser - Passed from parent container and has everything about the logged in user
 * @param {Array} courses - List of courses passed to this component from the db
 * @param {Number} points - Number of points the user has for this course
 */

import React from "react";

import { ReactComponent as Js } from "assets/js.svg";
import { ReactComponent as Html } from "assets/html.svg";
import { ReactComponent as ReactJsx } from "assets/react-jsx.svg";
import { ReactComponent as ReactStyle } from "assets/react-style.svg";

import Grid from "@material-ui/core/Grid";
import Grow from "@material-ui/core/Grow";
import LockIcon from "@material-ui/icons/Lock";
import MoCard from "components/library/MoCard";

const types = {
  js: () => <Js />,
  html: () => <Html />,
  reactJsx: () => <ReactJsx />,
  reactStyle: () => <ReactStyle />,
  disabled: () => <LockIcon />
};

const CoursesList = ({ authUser, courses, collectionPath, points }) => {
  return courses.map((course, index) => {
    /* Configure url route for each item */
    const configureUrl = course.isDisabled
      ? ""
      : `/${collectionPath}/${course.doc}`;

    const IconComponent = course.isDisabled
      ? types.disabled
      : types[course.type];

    return (
      <Grow in={course && true} key={index}>
        <Grid item xs={12} sm={12} md={4} lg={4}>
          <MoCard
            isDisabled={course.isDisabled}
            points={points}
            title={course.title}
            subtitle={course.subtitle}
            url={configureUrl}
            IconComponent={IconComponent}
          />
        </Grid>
      </Grow>
    );
  });
};

export default CoursesList;
