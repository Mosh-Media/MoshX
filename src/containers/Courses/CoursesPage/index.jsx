/**
 * Collections is a container that fetches firebase data using hooks and renders a list of all collections
 * @param {Object} authUser - Passed from parent container and has everything about the logged in user
 * @param {Object} collection - Passed from parent with a title and a path of the collection
 * @param {Object} firebase - Firebase class provides access to authUser and db - comes from withAuthentication hoc
 * @param {Object} match - Contains information about how a <Route path> matched the URL - comes from withRouter and passed to withAuthentication hoc
 * @returns {<CoursePage/>} - returns CoursePage component which contains the rest of the components
 */

import React from "react";

import { withAuthentication } from "components/shared/Session";
import CoursesList from "./components/CoursesList";
import Footer from "components/shared/Footer";
import Grid from "@material-ui/core/Grid";
import MoPage from "components/library/MoPage";
import useCollections from "hooks/useCollections";


const CoursesPage = ({
  authUser,
  collection,
  history,
  firebase,
}) => {

  const collectionDetails = {
    title: collection.title,
    isProgressBar: collection.isProgressBar,
    collectionPath: collection.path,
    locationHash: history.location.hash
  }

  const courses = useCollections(
    collectionDetails,
    firebase
  );


  if (!courses || !courses?.data?.length) {
    return null;
  }

  return (
    <MoPage title={collectionDetails?.title}>
      <Grid container spacing={4} alignItems="center">
        <CoursesList
          authUser={authUser}
          courses={courses.data}
          collectionPath={"courses"}
          points={0}
        />
      </Grid>
      <Footer />
    </MoPage>
  );
};

export default withAuthentication(CoursesPage);
