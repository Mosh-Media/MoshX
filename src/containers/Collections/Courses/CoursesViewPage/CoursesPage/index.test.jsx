import React from "react";

import ShallowRenderer from "react-test-renderer/shallow";
import CoursesPage from ".";

const renderer = new ShallowRenderer();

describe("CoursesPage Component", () => {
  test("should match the snapshot", () => {
    renderer.render(<CoursesPage />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
