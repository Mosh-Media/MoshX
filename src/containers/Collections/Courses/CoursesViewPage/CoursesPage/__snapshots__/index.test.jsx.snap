// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursesPage Component should match the snapshot 1`] = `
<ForwardRef(WithStyles)
  maxWidth="lg"
>
  <ForwardRef(WithStyles) />
  <ForwardRef(WithStyles)>
    <ForwardRef(Grow)
      in={true}
      mountOnEnter={true}
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <ForwardRef(WithStyles)
        alignItems="center"
        container={true}
        spacing={4}
      >
        <ForwardRef(WithStyles)
          item={true}
          lg={4}
          md={4}
          sm={12}
          xs={12}
        >
          <SignUpCard
            type="signup"
          />
        </ForwardRef(WithStyles)>
        <UNDEFINED />
      </ForwardRef(WithStyles)>
    </ForwardRef(Grow)>
    <ForwardRef(WithStyles) />
  </ForwardRef(WithStyles)>
</ForwardRef(WithStyles)>
`;
